src/t_set.c

```diff
@@ -525,15 +616,19 @@ void spopWithCountCommand(client *c) {
     if (remaining*SPOP_MOVE_STRATEGY_MUL > count) {
         while(count--) {
             /* Emit and remove. */
-            encoding = setTypeRandomElement(set,&sdsele,&llele);
+            encoding = setTypeRandomElement(set, &str, &len, &llele);
             if (encoding == OBJ_ENCODING_INTSET) {
                 addReplyBulkLongLong(c,llele);
                 objele = createStringObjectFromLongLong(llele);
                 set->ptr = intsetRemove(set->ptr,llele,NULL);
+            } else if (str == NULL) {
+                addReplyBulkLongLong(c,llele);
+                objele = createObject(OBJ_STRING, sdsfromlonglong(llele));
```

Comment:
```suggestion
                objele = createStringObjectFromLongLong(llele);
```

Reply:
`createStringObjectFromLongLong` returns a OBJ_ENCODING_INT with `shared.integers[llele]`, which is what caused some problems in the fuzz testing you helped me resolve. :-)